File: Leetcode/October/Day8_Binary_Search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int low = 0;
        int high = nums.size()-1;
        while(low <= high)
        {
             int mid = ((low + high) / 2);
            if(nums[mid] > target)
            {
                high = mid - 1;
            }
            else if(nums[mid] < target)
            {
                low = mid + 1;
            }
            else if(nums[mid] == target)
            {
                return mid;
            }
        }
        return -1;
    }
};

// Time Complexity - O(logn) (not nlogn because array already sorted)
// Space Complexity - O(1)

File: LeetCode/Monthly Challenges/October_LeetCoding_Challenge/Day8_Binary_Search/Day8BinarySearch.java
class Solution {
    public int search(int[] nums, int target) {
        
        int i = 0;
        int j = nums.length - 1;
        while(i <= j){
            int mid = (i+j)/2;
            if(nums[mid] == target)
                return mid;
            else if(nums[mid] > target){
                j--;
            }
            else
                i++;
        }
        return -1;
    }
}
